### PR TITLE
Add proposal side panel to organizations page

### LINKED ISSUE #613: Add proposal side panel to organization page
When users click on a proposal in the table on the org detail page, we should load it up into a third panel just like the current data providers data:

Org - Proposal

- Populate the panel just like it would be on the full-sized detail page
- X icon closes the panel
- Expand icon is a link to the proposal detail page itself (so, leave the "org" context entirely; literally just a route to `/proposals/1234` or whatever)
- Highlight current proposal in table (in the org panel)

### PR DESCRIPTION
**Pull request by hminsky2002**
This PR adds a side panel to the organizations detail page that is opened via clicking on a proposal in the Organizations proposal table. On click of the expansion icon, the user will be redirected to the related proposals detail page.

Closes #613 
Testing:
1. Start PDC front-end
2. Populate the database with (at least) 1 organization, 1 proposal, and 1 OrganizationProposal relating the organization and proposal
3. Navigate to `/organizations/1`
4. Click on the related proposal in the Organization's proposal table
5. Verify that the side panel opens on click, closes on click of the X icon, and redirects to the proposal's detail page on click of the expand icon

### CODE REVIEW

**Comment by reefdog** [comment]
I'm starting my review of this one! Can I make three quick suggestions before I get started:

- Do a typo/formatting pass on the commit messages, especially titles (see "[Add OrganzationProposal Panel](https://github.com/PhilanthropyDataCommons/front-end/pull/673/commits/44eeaf6f345ed23b4b94be53eaf755af52969e2c)").
- It looks like you ran Prettier on the final commit only, because it applied some formatting to new code from prior commits on this branch. Please do run Prettier on each commit individually; that way when I'm reviewing commit-by-commit, a later commit doesn't have any formatting-only noise.
- When you open one of these laddered PRs, set the base to the branch you laddered from (in this case, `612-proposals-table-org-page`). That way it will only show the diff commits. Once you merge that one in to `main`, GitHub will automatically change the base of this one to `main`.

File: src/components/OrganizationProposal/OrganizationProposalPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import { ClosablePanel, PanelTitle } from '../Panel';
+import { ProposalTable } from '../ProposalTable';
+import { Button } from '../Button';
+
+interface OrganizationProposalTitleProps {
+	proposalName: string;
+}
+const OrganizationProposalTitle = ({
+	proposalName,
+}: OrganizationProposalTitleProps) => <PanelTitle>{proposalName}</PanelTitle>;
```
[the remark this answers is not in the document]
Comment (hminsky2002) [reply src/components/OrganizationProposal/OrganizationProposalPanel.tsx:13]: Agreed on second look, definitely unnecessary!

File: src/components/ProposalListTable.tsx [hunk 2]
```diff
@@ -46,6 +54,8 @@ interface ProposalListTableProps {
 	proposals: FrontEndProposal[];
 	wrap?: boolean;
 	columns?: string[];
+	organization?: boolean;
+	organizationId?: number;
```
[the remark this answers is not in the document]
Comment (hminsky2002) [reply src/components/ProposalListTable.tsx]: The methodology I ended up using was introducing the enum! Take a look at the code itself, I agree there may be another way to tackle it but interested in your thoughts on this approach(code is contained in commit 2)!

File: src/components/Table/RowCell.tsx [hunk 3]
```diff
@@ -4,12 +4,18 @@ interface RowCellProps {
 	children?: React.ReactNode;
 	className?: string;
 	wrap?: boolean;
+	active?: boolean;
```
Comment (reefdog) [inline src/components/Table/RowCell.tsx:7]: Per the earlier comment about not needing `active` on a cell, let's revert the changes in this file.

File: src/components/OrganizationProposal/OrganizationProposalPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+interface OrganizationProposalPanelProps {
+	proposalId: string;
+	version: number;
+	values: {
+		shortCode: string;
+		fieldName: string;
+		position: number;
+		value: string;
+	}[];
+	onClose: () => void;
```
[the remark this answers is not in the document]
Comment (reefdog) [reply src/components/OrganizationProposal/OrganizationProposalPanel.tsx:39]: It's like you gave me a new toy and now I want it everywhere, lol.

File: src/components/Table/Table.css [hunk 5]
```diff
@@ -77,6 +77,14 @@
 	background-color: var(--color--gray--lighter);
 }
 
+.table tr.active {
+	background-color: var(--color--blue--lighter);
+}
+
+.table td.active {
```
[the remark this answers is not in the document]
Comment (hminsky2002) [reply src/components/Table/Table.css:84]: @reefdog 
noted! the only reason was because when the property was only present on the row, the initial click would not show the row turning blue, only after moving the cursor would it then be highlighted. Is that ok? I think it's because the styles on the cells override the row styles while it's on hover

[the remark this answers is not in the document]
Comment (hminsky2002) [reply src/components/Table/Table.css:84]: perhaps something like this could work?
```css
.table tr.active {
    background-color: var(--color--blue--lighter);

    td {
        background-color: var(--color--blue--lighter);
    }
}
```

[the remark this answers is not in the document]
Comment (reefdog) [reply src/components/Table/Table.css]: @hminsky2002 Ah right, the hovering rule beats it.

So first of all, you're right, we want the active state to beat the normal hover state. But let's do this instead:

```css
.table .clickable:not(.active):hover {
	background-color: var(--color--gray--lighter);
}

.table .active {
	background-color: var(--color--blue--lighter);
}
```

1. This way we're explicitly saying "we want clickable rows to have a hover state, but only when they're not active" which is the right level of specificity IMO
2. No need for the element names in the selectors (e.g., `tr` or `td`), we aren't meaningfully distinguishing based on element so it's unnecessary selector specificity

[the remark this answers is not in the document]
Comment (hminsky2002) [reply src/components/Table/Table.css]: @reefdog roger that! modified rebased and force pushed!

[the remark this answers is not in the document]
Comment (reefdog) [reply src/components/Table/Table.css]: Oh on another note: your block above would be the first time we use SASS-like CSS nesting. I adore CSS nesting and am so glad it's finally available natively, but it's only just available in all the evergreen browsers as of December 2023. It's perfectly okay to use it, especially as we're building a "tool for the future" and don't have real users to get in trouble with yet, but let's be especially sure to test any use of it across all target browsers (Chrome, Firefox, Safari) since it's still so new.

File: src/components/ProposalListTable.tsx [hunk 6]
```diff
@@ -5,25 +5,46 @@ import { getPreferredProposalApplicantNameValues } from '../utils/proposals';
 import { TableRow, RowCell } from './Table';
 import { ListTable } from './ListTable';
 
+export enum ProposalDetailDestinations {
```
[the remark this answers is not in the document]
Comment (reefdog) [reply src/components/ProposalListTable.tsx]: I accidentally copy-edited out the explanation of _why_ I ask this: so we can always rely on looking at the bottom of a file to see its exports. When the last declared variable is exported (and is the only one), that's easy, and when there's an export block at the bottom, that's easy. But when there are multiple exports sprinkled throughout, especially with sibling declarations that aren't exported, that makes it a little harder. So it's a method to help ourselves when reading a file.

File: src/components/Table/Table.css (hunk 5 above)
[the remark this answers is not in the document]
Comment (hminsky2002) [reply src/components/Table/Table.css]: @reefdog yeah it was a surprise to me that it accepted the nested definition in my interpreter, I had to do a double take! I was just writing it out as pseudocode but then voila. Very happy they're making that the standard.

File: src/components/Table/Table.css [hunk 7]
```diff
@@ -77,6 +77,14 @@
 	background-color: var(--color--gray--lighter);
 }
 
+.table .clickable:not(.active):hover {
+	background-color: var(--color--gray--lighter);
+}
```
Comment (reefdog) [inline src/components/Table/Table.css:82]: Ah you did good but this rule was meant to *replace* the one above it. Sorry, I should have been more clear!

So basically you need to remove lines 76–79 now.

Comment (hminsky2002) [reply src/components/Table/Table.css:78]: ah that was mindless of me! should be remedied now

File: src/components/Table/Table.css (hunk 5 above)
[the remark this answers is not in the document]
Comment (reefdog) [reply src/components/Table/Table.css]: @hminsky2002 It's _so wonderful_. With CSS variables, CSS nesting, `calc()`, and some of the new color methods, we're really in a golden age of native CSS and ditching preprocessed supersets like SCSS. There are still a few things they do that native CSS doesn't, but not nearly enough to justify their added build complexity IMO.

**Review by reefdog — APPROVED** [review]
Huzzah! Thanks for all the iterating.

**Comment by hminsky2002** [comment]
> Huzzah! Thanks for all the iterating.

Thanks for the reviews! This ones probably my coolest contribution yet, i enjoy opening and closing the panel immensely

**Comment by reefdog** [comment]
@hminsky2002 lol, me too.
